framework/src/actions/Action.C

```diff
@@ -173,3 +178,26 @@ Action::getBaseName() const
   mooseDeprecated("getBaseName() is deprecated.");
   return MooseUtils::baseName(_name);
 }
+
+void
+Action::connectControllableParams(const std::string & parameter,
+                                  const std::string & object_type,
+                                  const std::string & object_name,
+                                  const std::string & object_parameter) const
+{
+  MooseObjectParameterName primary_name(uniqueActionName(), parameter);
+  MooseObjectParameterName secondary_name(object_type, object_name, object_parameter);
```

Comment:
Can you pass in `object_params` instead of `object_type`? This is what we do in `Component`s in THM. Then you can get the base type with `object_params.get<std::string>("_moose_base")`.

Reply:
I think this is more straight-forward to developers. At the place this function is called, developers know exactly the object type, name and parameter. It will take them some thinking to pass in the MooseObjectName or MooseObjectParameterName.

Reply:
Oh, I see what you mean. Yeah, MOOSE base is a bit difficult for developers to understand. Ideally here we pass in the object type, but in reality here it is a MOOSE base name. Is there a function somewhere we can use to get the base from a type, like `giveMeBase("Diffusion")` returning `Kernel`?

Reply:
I will just pass in the parameter name. If we do this, we do not need the object name as well because it can be obtained from `_object_name`.